src/main/java/pro/java/hw17/FileLogger.java

```diff
@@ -0,0 +1,153 @@
+                String newLogFileMessage = "New log file created: " + currentFileName;
+                String newPathMessage = GREEN + "Path to the new log file: " + ARROW + newPath.toAbsolutePath() + RESET;
+                throw new FileMaxSizeReachedException(errorMessage, limitMessage, beforeRotationMessage, newLogFileMessage, newPathMessage);
+            } catch (IOException e) {
+                String errorMessage = "Error creating a new log file: " + e.getMessage();
+                String limitMessage = "Log file size limit: " + config.getMaxSize() + " bytes";
+                String beforeRotationMessage = "Log file size before creating a new one: " + fileSizeBeforeRotation + " bytes";
+                String failedLogMessage = "Failed to create a new log file.";
+                String failedLogPathMessage = GREEN + "Path to the failed log file: " + ARROW + Paths.get(currentFileName).toAbsolutePath() + RESET;
+                throw new FileMaxSizeReachedException(errorMessage, limitMessage, beforeRotationMessage, failedLogMessage, failedLogPathMessage);
+            }
+        }
+    }
+
+    private String getTimestamp() {
+        SimpleDateFormat dateFormat = new SimpleDateFormat(config.getLogFormat());
+        return dateFormat.format(new Date());
+    }
+
+    private void log(String message, LoggingLevel level) throws FileMaxSizeReachedException, IOException {
+        if (level.ordinal() <= config.getLogLevel().ordinal()) {
+            checkFileSize();
+            SimpleDateFormat dateFormat = new SimpleDateFormat(config.getLogFormat());
+            String timestamp = dateFormat.format(new Date());
+            String logMessage = String.format("[%s][%s] %s: %s", timestamp, level, "Message", message);
+            writer.println(logMessage);
+            writer.flush();
+            currentFileSize += logMessage.length();
+        } else {
+            System.out.println("Log message is not logged due to wrong logging level: " + level);
+        }
+    }
+
+    public void warning(String message) {
+        try {
+            if (config.isLogToConsole()) {
+                String logMessage = "[" + getTimestamp() + "][" + LoggingLevel.WARN + "] " + "Message: " + message;
+                System.out.println(logMessage);
+            } else {
+                log(message, LoggingLevel.WARN);
+            }
+        } catch (FileMaxSizeReachedException e) {
+            displayFileMaxSizeMessages(e);
+        } catch (IOException e) {
+            System.err.println("Error occurred while logging WARNING message: " + message);
+        }
+    }
+
+    public void info(String message) {
+        if (LoggingLevel.INFO.ordinal() <= config.getLogLevel().ordinal()) {
+            try {
+                if (config.isLogToConsole()) {
+                    String logMessage = "[" + getTimestamp() + "][" + LoggingLevel.INFO + "] " + "Message: " + message;
```

Comment:
You can move this logic to the method just to avoid a duplication:
void print (boolean isLogToConsole, LoggingLevel) {
            if (config.isLogToConsole()) {
                String logMessage = "[" + getTimestamp() + "][" + LoggingLevel + "] " + "Message: " + message;
                System.out.println(logMessage);
            } else {
                log(message, LoggingLevel);
            }
}

And call it from info, debug and other methods

Reply:
fixed